mteb/tasks/Retrieval/code/WikiSQLRetrieval.py

```diff
@@ -0,0 +1,88 @@
+from __future__ import annotations
+
+from mteb.abstasks.TaskMetadata import TaskMetadata
+
+from ....abstasks.AbsTaskRetrieval import AbsTaskRetrieval
+
+
+class WikiSQLRetrieval(AbsTaskRetrieval):
+    metadata = TaskMetadata(
+        name="WikiSQLRetrieval",
+        description="A code retrieval task based on WikiSQL dataset with natural language questions and corresponding SQL queries. Each query is a natural language question (e.g., 'What is the name of the team that has scored the most goals?'), and the corpus contains SQL query implementations. The task is to retrieve the correct SQL query that answers the natural language question. Queries are natural language questions while the corpus contains SQL SELECT statements with proper syntax and logic for querying database tables.",
+        reference="https://huggingface.co/datasets/embedding-benchmark/WikiSQL",
+        dataset={
+            "path": "embedding-benchmark/WikiSQL",
+            "revision": "2aee0c8bf3e5e1d92af8e6998d7c6f31bda95e6f",
+        },
+        type="Retrieval",
+        category="s2s",
+        modalities=["text"],
+        eval_splits=["test"],
+        eval_langs=["eng-Latn", "sql-Code"],
+        main_score="ndcg_at_10",
+        date=("2017-01-01", "2017-12-31"),
+        domains=["Programming"],
+        task_subtypes=["Code retrieval"],
+        license="bsd-3-clause",
+        annotations_creators="expert-annotated",
+        dialect=[],
+        sample_creation="found",
+        bibtex_citation="""@article{zhong2017seq2sql,
+  archiveprefix = {arXiv},
+  author = {Zhong, Victor and Xiong, Caiming and Socher, Richard},
+  eprint = {1709.00103},
+  primaryclass = {cs.CL},
+  title = {Seq2SQL: Generating Structured Queries from Natural Language using Reinforcement Learning},
+  year = {2017}
+}""",
```

Comment:
tests seem to fail due to misformatted bibtex (you can simply format that using `make format-citations`